nixos/modules/services/logging/packetbeat.nix

```diff
@@ -0,0 +1,269 @@
+                  <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/configuration-flows.html'/>
+                  for all available configuration options.
+                '';
+              };
+
+              protocols = mkOption {
+                type = types.listOf format.type;
+                description = ''
+                  Configuration of what protocols packetbeat should gather info about.
+                  See <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/configuration-protocols.html'/>
+                  for the configuration options available.
+                '';
+              };
+            };
+            output = mkOption {
+              type = format.type;
+              description = ''
+                Configuration of where packetbeat should send the information its gathered.
+                See <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/configuring-output.html'/>
+                for the configuration options available.
+              '';
+            };
+
+            setup = mkOption {
+              type = format.type;
+              description = ''
+                Configures where the kibana endpoint is for setting up dashboards and
+                whether packetbeat should setup ILM and index templates in elasticsearch.
+              '';
+            };
+
+            fields = mkOption {
+              type = format.type;
+              description = ''
+                Extra fields that packetbeat should add to the top-level of the objects it
+                sends to its output.
+              '';
+            };
+
+            processors = mkOption {
+              type = format.type;
+              description = ''
+                Configuration of what enrichment and/or filtering processors packetbeat
+                should gather information from before sending it to the output.
+                See <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/filtering-and-enhancing-data.html'/>
+                for the configuration options available.
+              '';
+            };
+          };
+        };
+        default = {};
+        description = ''
+          Any other configuration options you want to add.
+        '';
+        example = ''
+          output = {
+            elasticsearch = {
+              hosts = [ "localhost:9200" ];
+            };
+          };
+          setup = {
+            template = {
+              settings = {
+                index.number_of_shards =  1;
+              };
+            };
+            kibana = {
+              host =  "localhost:5601";
+            };
+          };
+          processors =  [
+            '''
+              if.contains.tags: forwarded
+              then:
+                - drop_fields:
+                  fields: [host]
+              else:
+                - add_host_metadata: ~
```

Comment:
Is this yaml in a string? Shouldn't this be turned into the Nix equivalent?

Reply:
This is indeed YAML in a string, and the reason why its YAML in a string, is that I have no idea how to turn it into the nix equivalent, mainly because I'm not entirely sure what YAML its actually using. It looks like its a legit expression, pythonesque. I can see that the statements after then and else are objects, but how I would represent the actual expression I don't know.

Reply:
This should be solved now, made them into nested attrsets